Assets/Scripts/Models/Buildable/Components/Component.cs

```diff
@@ -0,0 +1,119 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+
+namespace ProjectPorcupine.Buildable.Components
+{
+    [Serializable]
+    public abstract class Component
+    {
+        protected static readonly string ComponentLogChannel = "FurnitureComponents";
+
+        private static Dictionary<string, Type> componentTypes;
+        
+        public Component()
+        {
+            // need to set it, for some reason GetHashCode is called during serialization (when Name is still null)
+            Type = string.Empty;
+        }
+        
+        [XmlIgnore]
+        public Furniture ParentFurniture { get; set; }
```

Comment:
I think that every component should link to a Buildable, so that we can use some components for both Furniture, Utility and other buildable clases. And we should replace places where Furniture is used (as comments, part of Function names...) to Buildable or just remove it.